zerver/lib/events.py

```diff
@@ -212,6 +212,14 @@ def fetch_initial_state_data(user_profile: UserProfile,
         state['email'] = user_profile.email
         state['full_name'] = user_profile.full_name
 
+        # Set value for custom user profile fields
+        for field in custom_profile_fields_for_realm(user_profile.realm.id):
+            try:
+                value = CustomProfileFieldValue.objects.get(user_profile=user_profile, field=field).value
+            except CustomProfileFieldValue.DoesNotExist:
+                value = ""
+            state[field.name] = value
```

Comment:
This fetch the user profile values for custom fields. If user hasn't set any value for custom fields yet, it will pass `""`.